tests/migration/reset_scc_addons_values.pm

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Swap the values of SCC_ADDONS before migration.
+#
+# In migration maintenance updates tests, SCC_ADDONS is used for both
+# registering original system and adding target system modules maintenance
+# updates repos. But for SLE 12 and SLE 15, the modules name are different.
+# So swap SCC_ADDONS values to upgraded SLE 15 modules before migration.
+# And re-create incidents repos based on the swapped SCC_ADDONS values.
+#
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    select_console 'root-console';
+    set_var('SCC_ADDONS', get_var('SCC_ADDONS_2'));
+    # Recreate the incidents repos based on the swapped SCC_ADDONS values.
+    my ($maint_test_repos, $test_repo) = ('', '');
```

Comment:
There is something peculiar in Perl that you can use to avoid to control so much the values and add condition code.

Pushing an empty array into an array doesn't create any undef or gaps in the list:
```
my @maint_test_repo;
push(@maint_test_repo, ());
print Dumper(@maint_test_repo), "\n";
```
You can try this snippet:
```
#!/usr/bin/perl
use strict;
use warnings;
use Data::Dumper;

sub get_repo {
  my $addon = shift;
  return "http://repo/with/patch/for/base/1,http://repo/with/patch/for/base/2" if $addon eq "base";
  return "http://repo/with/patch/for/desktop/1" if $addon eq "desktop";
  return "http://repo/with/patch/for/contm/1,http://repo/with/patch/for/contm/2,http://repo/with/patch/for/contm/3 " if $addon eq "contm";
  return "";
}

sub get_addons {
  return "base,serverapp,desktop,sdk,legacy,python3,script,contm,pcm,we";
}

my @maint_test_repo = ();
for my $addon (split(/,/, get_addons())) {
  my @result = split(/,/, get_repo($addon));
  push(@maint_test_repo, split(/,/, get_repo($addon)));
}

push(@maint_test_repo, ());
print Dumper(@maint_test_repo), "\n";
```
that means that doing something similar and using the default parameter of get_var it should simplify the code: `get_var(uc($addon) . '_TEST_REPOS', "")`